lib/Sema/CSDiagnostics.h

```diff
@@ -95,10 +95,27 @@ class FailureDiagnostic {
   Type getType(Expr *expr, bool wantRValue = true) const;
   Type getType(const TypeLoc &loc, bool wantRValue = true) const;
 
+  Type simplifyType(Type type) const  {
```

Comment:
Let's not add `simplifyType` to a `FailureDiagnostic`, let's just inline the call to `simplifyTypeImpl` there. It would be easier there was only one way to "resolve" a type.